### DISCUSSION TITLE
Measuring Distance Travelled Using a Reference

### DISCUSSION THREAD

**Discussion by Jaspermoray** [talmolab/sleap#1510] [General]
Hi all!

I'm a little stuck in data analysis here. I'm trying to figure out how to measure the total distance in centimeters an animal travels. I understand that to do this, you need a reference that you can measure and get a conversion factor from pixels in a frame to cm. But I can't figure out how to do it in practice. Does anybody have ideas?

Best,
Jasper

**Comment by Alan-Marcus** [comment]
Hi Jasper,
To obtain the video to real-world conversion factor:
- have an item of known of known size in your video (e.g. length of feeding tray in cm);
- determine the coordinates of each end of this item: add a single-node instance to either end of the item (or an instance with two nodes);
- obtain these coordinates (I exported the data to CSV; depends on your workflow);
- calculate the pixel length between nodes: SQRT((x2-x1)^2+(y2-y1)^2);
- calculate the pixels per cm: "pixel length" / "object length in cm".

Next you want to determine the pixel distance your animal (node) moves between frames, sum this distance, then use your above conversion factor to determine actual distance. There are a number of steps and considerations involved here - how to deal with missing data; smoothing / thresholding the data; use every frame or every nth frame? etc... Do you need help with this or just the above video to real-world conversion?

You can also see what I did here: https://doi.org/10.1016/j.xpro.2022.101326

Cheers,
Alan

**Reply by sronilsson** [reply]
As above! Some methods documented [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/Tutorial_tools.md#get-coordinates-calibrate-distance) or [HERE](https://github.com/sgoldenlab/simba/blob/master/docs/Scenario1.md#step-3-set-video-parameters) and code [here](https://github.com/sgoldenlab/simba/blob/master/simba/video_processors/calculate_px_dist.py)

**Reply by Jaspermoray** [reply]
Thank you both so much! I'll work on these and let you know if I have questions.
